src/Http/Headers/src/HeaderUtilities.cs

```diff
@@ -566,18 +515,17 @@ public unsafe static string FormatNonNegativeInt64(long value)
             }
 
             var position = _int64MaxStringLength;
-            char* charBuffer = stackalloc char[_int64MaxStringLength];
+            Span<char> charBuffer = stackalloc char[_int64MaxStringLength];
```

Comment:
string.Create + https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.Private.CoreLib/src/System/Buffers/Text/FormattingHelpers.CountDigits.cs#L13 ?

Reply:
I wonder if we can just use value.ToString now. @stephentoub has it been optimized enough at this point to nuke this method?

Reply:
`return ((ulong)value).ToString();` will give you the best possible performance.

Reply:
Thanks for confirming!

Reply:
What Levi said.

Reply:
Thanks for double confirming. BTW I hate that we still have these strongly typed headers as part of ASP.NET Core. In some distant future I'd like to see these dead and we can just consume the parsers directly from the BCL.